src/main/java/org/onedatashare/transferservice/odstransferservice/service/cron/MetricsCollector.java

```diff
@@ -85,44 +113,56 @@ public void collectAndSave() {
         if (pmeterMetrics.size() < 1) return;
 
         this.previousParentMetric = influxCache.someJobMetric(); //this metrics throughput is the throughput of the whole map in influxCache.
-        long jobSize = 0L;
-        long avgFileSize = 0L;
-        long pipeSize = 0L;
+
         String destType = "";
         String sourceType = "";
         String ownerId = this.odsUser;
+
         if (this.previousParentMetric.getStepExecution() != null) {
             JobParameters jobParameters = this.previousParentMetric.getStepExecution().getJobParameters();
-            jobSize = jobParameters.getLong(JOB_SIZE);
-            avgFileSize = jobParameters.getLong(FILE_SIZE_AVG);
-            pipeSize = jobParameters.getLong(PIPELINING);
+            jobSize.set(jobParameters.getLong(JOB_SIZE));
+            avgFileSize.set(jobParameters.getLong(FILE_SIZE_AVG));
+            pipeSize.set(jobParameters.getLong(PIPELINING));
+            throughput.set(this.previousParentMetric.getWriteThroughput());
```

Comment:
So this value should be recorded separately maybe from read() and write().
The following approaches u can take which should be acceptable are:
the min(readThroughput, writeThroughput). 
Or keep track of throughput entirely